include/swift/Runtime/Concurrency.h

```diff
@@ -107,6 +107,10 @@ void swift_task_dealloc(AsyncTask *task, void *ptr);
 SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
 void swift_task_cancel(AsyncTask *task);
 
+/// Get 'active' AsyncTask, depending on platform this may use thread local storage.
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+AsyncTask* swift_task_get_active();
```

Comment:
If we do it this way things work fine.

I am confused how to make a Builtin which will work the same way.
When using a builtin we end up with unnecessary retain/releases I think, see the bug that the naive impl of current would cause here: https://github.com/apple/swift/pull/36119#issuecomment-784800771

cc @DougGregor

I know we do `CONSTANT_OWNERSHIP_BUILTIN(None, GetCurrentAsyncTask)` for the getCurrentAsyncTask but AFAICS that causes wrong retain/release things to the returned task. Whereas returning here does not.

This is a bit beyond me which "way" is "right" though -- should this be a builtin, or rather, replace the `getCurrentAsyncTask`? But then we need to fix those wrong retain/releases it causes.